--no-js-ffi opt to disable JS FFI mangling.

kanaka opened the pull request:
For JS/Web platform, calls to JS imports are wrapped to convert i64 to
i32 and f32 to f64. Likewise calls from JS into exports do the inverse
wrapping. However, for non-web embeddings this is probably undesirable.
This change provides an option to disable that wrapping and use the
original types for the call.

To fully disable JS FFI mangling, fastcomp FFI mangling must also be
disabled using the --emscripten-js-ffi=0 flag (https://github.com/kripken/emscripten-fastcomp/pull/181). It is used by emscripten using the JS_FFI flag (https://github.com/kripken/emscripten/pull/5168).

kripken commented on src/asm2wasm.h:
```diff
@@ -342,6 +342,7 @@ class Asm2WasmBuilder {
     
   TrapMode trapMode;
   PassOptions passOptions;
+  bool runJsFfiPass;
```
how about renaming to `legalizeJSFFI`?

kripken replied:
Or maybe `legalizeJavaScriptFFI`

kanaka replied:
That's fine with me. Do you want me to use a similar name for the emscripten and fastcomp options? "legalize" seemed a bit specific to binaryen so that's why I moved towards a more generic naming.

kripken commented:
Overall seems reasonable. We should add a test for this, though. Need to modify `check.py` and `auto_update_tests.py`, look for existing tests with asm2wasm, hopefully there's an obvious way to add it there. Maybe similar to how we test the `-g` flag.

kanaka commented:
Okay, I'll look into adding a test.

kanaka commented:
I've updated the PRs with the new parameter names. I've also added a test here that makes sure the output is correct (no type mangling/legalizing) when the option is used.

kripken commented:
Great, thanks.
